q2lint.py

```diff
@@ -38,9 +38,14 @@ def main():
 
         for package in npm_packages:
             pkg_path = package.parent
-            cmd = 'cd %s && npm i && npm run build -- --bail && (git diff ' \
+            install = 'cd %s && npm i --silent --progress false' % pkg_path
+            cmd = subprocess.run(install, shell=True,
+                                 stdout=subprocess.DEVNULL)
+            if cmd.returncode != 0:
+                errors.append('npm install failed')
+            cmd = 'cd %s && npm run build -- --bail && (git diff ' \
                   '--quiet */bundle.js || bash -c "exit 42")' % pkg_path
-            res = subprocess.run(cmd, shell=True)
+            res = subprocess.run(cmd, shell=True, stdout=subprocess.DEVNULL)
             if res.returncode == 42:
                 errors.append('Bundle is out of sync for %s' % pkg_path)
             elif res.returncode != 0:
```

Comment:
I can't comment on L53, but when we add the `npm build error`, can we stop including the `res.stderr`? This is what it looks like right now:

```bash
(q2dev-2017.6) ☁  q2-taxa [master] ⚡ rm -rf q2_taxa/assets/barplot/node_modules && python ../q2lint/q2lint.py
Error
    at NormalModule.onModuleBuildFailed (/Users/matthew/caporaso/qiime2/q2-taxa/q2_taxa/assets/barplot/node_modules/webpack-core/lib/NormalModuleMixin.js:315:19)
    at nextLoader (/Users/matthew/caporaso/qiime2/q2-taxa/q2_taxa/assets/barplot/node_modules/webpack-core/lib/NormalModuleMixin.js:270:31)
    at /Users/matthew/caporaso/qiime2/q2-taxa/q2_taxa/assets/barplot/node_modules/webpack-core/lib/NormalModuleMixin.js:292:15
    at runSyncOrAsync (/Users/matthew/caporaso/qiime2/q2-taxa/q2_taxa/assets/barplot/node_modules/webpack-core/lib/NormalModuleMixin.js:173:4)
    at nextLoader (/Users/matthew/caporaso/qiime2/q2-taxa/q2_taxa/assets/barplot/node_modules/webpack-core/lib/NormalModuleMixin.js:290:3)
    at /Users/matthew/caporaso/qiime2/q2-taxa/q2_taxa/assets/barplot/node_modules/webpack-core/lib/NormalModuleMixin.js:292:15
    at Object.context.callback (/Users/matthew/caporaso/qiime2/q2-taxa/q2_taxa/assets/barplot/node_modules/webpack-core/lib/NormalModuleMixin.js:148:14)
    at Object.module.exports (/Users/matthew/caporaso/qiime2/q2-taxa/q2_taxa/assets/barplot/node_modules/eslint-loader/index.js:207:8)
    at WEBPACK_CORE_LOADER_EXECUTION (/Users/matthew/caporaso/qiime2/q2-taxa/q2_taxa/assets/barplot/node_modules/webpack-core/lib/NormalModuleMixin.js:155:71)
    at runSyncOrAsync (/Users/matthew/caporaso/qiime2/q2-taxa/q2_taxa/assets/barplot/node_modules/webpack-core/lib/NormalModuleMixin.js:155:93)
    at nextLoader (/Users/matthew/caporaso/qiime2/q2-taxa/q2_taxa/assets/barplot/node_modules/webpack-core/lib/NormalModuleMixin.js:290:3)
    at /Users/matthew/caporaso/qiime2/q2-taxa/q2_taxa/assets/barplot/node_modules/webpack-core/lib/NormalModuleMixin.js:259:5
    at Storage.finished (/Users/matthew/caporaso/qiime2/q2-taxa/q2_taxa/assets/barplot/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:38:16)
    at /Users/matthew/caporaso/qiime2/q2-taxa/q2_taxa/assets/barplot/node_modules/graceful-fs/graceful-fs.js:78:16
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:416:3)
Error
    at transpile (/Users/matthew/caporaso/qiime2/q2-taxa/q2_taxa/assets/barplot/node_modules/babel-loader/lib/index.js:61:13)
    at Object.module.exports (/Users/matthew/caporaso/qiime2/q2-taxa/q2_taxa/assets/barplot/node_modules/babel-loader/lib/index.js:163:20)

npm ERR! Darwin 16.4.0
npm ERR! argv "/usr/local/Cellar/node/7.5.0/bin/node" "/usr/local/bin/npm" "run" "build" "--" "--bail"
npm ERR! node v7.5.0
npm ERR! npm  v4.1.2
npm ERR! code ELIFECYCLE
npm ERR! @ build: `webpack "--bail"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @ build script 'webpack "--bail"'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the  package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     webpack "--bail"
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/matthew/caporaso/qiime2/q2-taxa/q2_taxa/assets/barplot/npm-debug.log


npm build error on q2_taxa/assets/barplot
None
```

It seems like the `None` at the very end might be confusing, and the error is already emitted above, so dropping `res.stderr` should be fine (this tool is for us, and for tracking down issues with PRs). I think just telling the dev `npm build error on %s' % pkg_path` is plenty!